dsn.go

```diff
@@ -684,6 +689,8 @@ func parseDSNParams(cfg *Config, params string) (err error) {
 			}
 		case "tracing":
 			cfg.Tracing = value
+		case "tmpDirPath":
+			cfg.TmpDirPath = value
```

Comment:
do we need to validate the value and throw an error if it's not valid such as empty, null, or doesn't exist? Also why we need this fetaure and not use the OS specific environment variable exist such as TMP or TMPDIR etc?

Reply:
Hey @sfc-gh-igarish, there are some details in #832 and #700, but having an OS-specific path causes a lot of confusion for end users who leverage this driver as part of more generic tooling. Where I work, Kubernetes clusters tend to be quite locked down and the default is to have a read-only file system, which caused some friction for people who tried to run this driver as part of a [Benthos](https://benthos.dev) pipeline. The first issue is that the package docs don't mention anything about this driver needing access to a tempdir to write data, so users found out the hard way by getting a cryptic panic (now fixed by emitting a more informative error). Then it's difficult to document in a clear way and it's not possible to expose it as something like `SNOWFLAKE_TMPDIR`, so users know exactly which temp files are produced by this driver, assuming the larger codebase which includes it might also need to write some temp files for other purposes. Finally, while I haven't studied the code in detail, I think it would be very useful for this driver to document explicitly in which situations it needs to write temp files and to provide a mechanism to supress them for simple workflows where the user just needs to stream some in-memory data to / from a stage file or to execute some simple SQL query. Hope that helps!

Reply:
@sfc-gh-pfus ok. please merge it. But still I don't understand why someone make TMP_DIR as read only. Temp directory meant for outputting any temporary files. Also advantage of default TEMP_DIR, for any reason file left over in it then on restart it can be clean easily and normally no one hesitate to clean the TMP_DIR as they know default usage of it when disk become full.